src/iter/find.rs

```diff
@@ -84,6 +84,14 @@ impl<'p, T, P> Folder<T> for FindFolder<'p, T, P>
         self
     }
 
+    fn consume_iter<I>(mut self, iter: I) -> Self where I: IntoIterator<Item=T> {
```

Comment:
Hmm. I'm not sure about this method. Two things:

- First, I don't know if it will be faster than the default impl. That default impl already just iterates and invokes `consume` until `full()` is true -- this is basically the same thing we are doing here.
- Second, in this version, we are not checking for the possibility that `self.found` gets set to true by somebody else, so we can wind up doing quite a bit more work. 

To fix the second part, you already can't use the built-in iterator `find` operation, since we want to be checking for `self.found` being set to true as well. Once we make those changes, I'm even less sure this will pay off. I think i'd be inclined *not* to include `consume_iter` here (and perhaps to leave a comment, instead, explaining that it likely doesn't pay off here).

Or am I missing something? (cc @cuviper -- thoughts?)

Reply:
I agree, this seems like a case that shouldn't be changed.  We want `find` to short-circuit.